format/macho/macho.go

```diff
@@ -0,0 +1,492 @@
+				return d.RawLen(int64((nmodules / 8) + (nmodules % 8)))
+			}) // TODO this needs better representation
+			d.FieldStrFn("name", func(d *decode.D) string {
+				return string(d.BytesLen(int(cmdsize) - int(offset)))
+			}) // TODO visualize this bitset
+		case LC_THREAD, LC_UNIXTHREAD:
+			d.FieldU32("flavor")
+			count := d.FieldU32("count")
+			d.FieldRawLen("state", int64(count*32))
+			// better visualization needed for this specific for major architectures
+		case LC_ROUTINES, LC_ROUTINES_64:
+			if archBits == 32 {
+				d.FieldU32("init_address")
+				d.FieldU32("init_module")
+				d.FieldU32("reserved1", d.BitBufIsZero())
+				d.FieldU32("reserved2", d.BitBufIsZero())
+				d.FieldU32("reserved3", d.BitBufIsZero())
+				d.FieldU32("reserved4", d.BitBufIsZero())
+				d.FieldU32("reserved5", d.BitBufIsZero())
+				d.FieldU32("reserved6", d.BitBufIsZero())
+			} else {
+				d.FieldU64("init_address")
+				d.FieldU64("init_module")
+				d.FieldU64("reserved1", d.BitBufIsZero())
+				d.FieldU64("reserved2", d.BitBufIsZero())
+				d.FieldU64("reserved3", d.BitBufIsZero())
+				d.FieldU64("reserved4", d.BitBufIsZero())
+				d.FieldU64("reserved5", d.BitBufIsZero())
+				d.FieldU64("reserved6", d.BitBufIsZero())
+			}
+		case LC_SUB_UMBRELLA, LC_SUB_LIBRARY, LC_SUB_CLIENT, LC_SUB_FRAMEWORK:
+			offset := d.FieldU32("offset")
+			d.FieldStrFn("name", func(d *decode.D) string {
+				return string(d.BytesLen(int(cmdsize) - int(offset)))
+			})
+		case LC_SYMTAB:
+			d.FieldU32("symoff")
+			d.FieldU32("nsyms")
+			d.FieldU32("stroff")
+			d.FieldU32("strsize")
+		case LC_DYSYMTAB:
+			// TODO
+			// if archBits == 32 {
+			// 	d.FieldStruct("nlist", func(d *decode.D) {
+			// 		d.FieldStruct("n_un", func(d *decode.D) {
+			// 			d.FieldS32("n_strx")
+			// 		})
+			// 		d.FieldU8("n_type")
+			// 		d.FieldU8("n_sect")
+			// 		d.FieldU16("n_desc")
+			// 		d.FieldU32("n_value")
+			// 	})
+			// } else {
+			// 	d.FieldStruct("nlist", func(d *decode.D) {
+			// 		d.FieldStruct("n_un", func(d *decode.D) {
+			// 			d.FieldS32("n_strx")
+			// 		})
+			// 		d.FieldU8("n_type")
+			// 		d.FieldU8("n_sect")
+			// 		d.FieldU16("n_desc")
+			// 		d.FieldU64("n_value")
+			// 	})
+			// }
+			d.FieldU32("ilocalsym")
+			d.FieldU32("nlocalsym")
+			d.FieldU32("iextdefsym")
+			d.FieldU32("nextdefsym")
+			d.FieldU32("iundefsym")
+			d.FieldU32("nundefsym")
+			d.FieldU32("tocoff")
+			d.FieldU32("ntoc")
+			d.FieldU32("modtaboff")
+			d.FieldU32("nmodtab")
+			d.FieldU32("extrefsymoff")
+			d.FieldU32("nextrefsyms")
+			d.FieldU32("indirectsymoff")
+			d.FieldU32("nindirectsyms")
+
+			d.FieldU32("extreloff")
+			d.FieldU32("nextrel")
+			d.FieldU32("locreloff")
+			d.FieldU32("nlocrel")
+		case LC_BUILD_VERSION:
+			d.FieldU32("platform")
+			d.FieldU32("minos")
+			d.FieldU32("sdk")
+			ntools := d.FieldU32("ntools")
+			var ntoolsIdx uint64
+			d.FieldStructArrayLoop("tools", "tools", func() bool {
```

Comment:
Second argument is name of struct in array (only shown with verbose display atm). I've usually used plural for the array and singular for each entry. So "tools" and "tool" here for example. But will be a lot of "tool" fiends here, could the "tool"  fiend in the struct be name something else, "name" etc?

Reply:
Adopting the plural singular scheme.